mysite/mysite/urls.py

```diff
@@ -35,6 +42,10 @@
     url(r'^set-pass/$', 'psa.views.set_pass'),
 
     url(r'^done/$', 'psa.views.done'),
+
+    url(r'^api/courses/(?P<course_id>\d+)/responses/$', ResponseViewSet.as_view({'get': 'list'}), name='responses'),
+    url(r'^api/courses/(?P<course_id>\d+)/errors/$', ErrorViewSet.as_view({'get': 'list'}), name='errors'),
+    url(r'^', include(router.urls)),
```

Comment:
Correct me if I am wrong, Here I didn't understand why we need `routers` as every viewset we are expecting url pattern should be `api/courses/(?P<course_id>\d+)/responses/` i.e. course_id should be in the url.
But here as you define `api/courses/errors` the matching request will also go to the same ResponseViewSet with no course_id. As a result it will show all model object.